### PR TITLE
feat(destination): Add new task manager adapter with channel driver

### PR DESCRIPTION
**Pull request by mohsenHa**
A new task manager adapter that uses the channel module to connect to the rabbitmq was added.

### CODE REVIEW

File: cmd/destination/main.go [hunk 1]
```diff
@@ -70,8 +73,32 @@ func main() {
 	// destination type coordinator which means every task with specific destination type
 	// will be published to its corresponding task publisher.
 
+	reconnectSecond := 5
+	channelSize := 100
+	numberInstant := 5
+	maxRetryPolicy := 5
```
Comment (iam-benyamin) [inline cmd/destination/main.go:79]: could you explain the retry policy in the destination I don't remember such a concept in destination

**Review by iam-benyamin — COMMENTED** [review]
thanks

File: cmd/destination/main.go (hunk 1 above)
Comment (mohsenHa) [reply cmd/destination/main.go:79]: It is used in the channel package. When a message is published to the input channel of the Rabbitmq adapter if it`s failed then retry until it is rich to the maxRetryPolicy.
We could add a delay time for retry in the future.

Comment (iam-benyamin) [reply cmd/destination/main.go:79]: thanks

**Review by gocastsian — APPROVED** [review]

**Pull request merged**